### PR TITLE
Generation

### PR DESCRIPTION
**Pull request by joshmoore**
This PR contains updates to the file generation logic to match the goals of #5 and #6. Additionally, it uses the work on nested storage from jzarr to deal with too many chunk files in a single directory.

### CODE REVIEW

**Comment by joshmoore** [comment]
Leaning towards getting this in so we can open more targeted follow-up PRs.

**Comment by joshmoore** [comment]
With the separation of setup/run in the previous commit, I now at least don't get Zarr _faster than_ 1-byte look up. fsspec is clearly doing a lot of heavy lifting.

image

**Comment by joshmoore** [comment]
Working locally for @will-moore . Merging for further refining.

File: docker-compose.yml [hunk 1]
```diff
@@ -60,8 +110,8 @@ services:
   minio4:
     image: minio/minio:RELEASE.2021-01-05T05-22-38Z
     volumes:
-      - data4-1:/data1
-      - data4-2:/data2
+      - "$ROOT}/data4-1:/data1"
+      - "$ROOT}/data4-2:/data2"
```
Comment (will-moore) [inline docker-compose.yml:114]: Typo on these 2 rows. Creates dir with `}`.

Comment (joshmoore) [reply docker-compose.yml:114]: Thanks, @will-moore. I'll push a fix.

**Pull request merged**